spicy/toolchain/src/compiler/codegen/parsers/literals.cc

```diff
@@ -163,11 +162,72 @@ struct Visitor : public hilti::visitor::PreOrder<Expression, Visitor> {
                 return dst;
             }
 
-            case LiteralMode::Try:
-                // RegExp are special-cased by the parser generator and handled
-                // directly there, so that we should get here in try mode.
-                hilti::logger().internalError("unexpected literal mode Try when parsing regexp literals");
-            default: hilti::util::cannot_be_reached();
+            case LiteralMode::Search: // Handled in `parseLiteral`.
+            case LiteralMode::Try: {
+                auto result = builder()->addTmp("result", state().cur);
```

Comment:
This identifier can clash with identifiers of the same name created higher up in the scope due to #1107. Let's use another name instead.